packages/node/src/handlers.ts

```diff
@@ -410,8 +410,9 @@ export function requestHandler(
           .then(() => {
             _end.call(this, chunk, encoding, cb);
           })
-          .then(null, e => {
+          .catch(e => {
```

Comment:
We can go back to `then(null, e`

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

> The catch() method returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) and deals with rejected cases only. It behaves the same as calling [Promise.prototype.then(undefined, onRejected)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) (in fact, calling obj.catch(onRejected) internally calls obj.then(undefined, onRejected)).

Reply:
And it actually should be this way, because `flush` returns a `PromiseLike` rather than a `Promise`, the difference being that only the latter has a `catch` method. (See https://github.com/microsoft/TypeScript/blob/f82d0cb3299c04093e3835bc7e29f5b40475f586/lib/lib.es5.d.ts#L1433-L1461.)

Reply:
Updated -- thanks for the additional context and clarity!